### PR TITLE
DS-4133 Upgrade Ruby to 3.3

### PR DESCRIPTION
**Pull request by ildomar-grings**
## Overview

[DS-4133](https://loyaltynz.atlassian.net/browse/DS-4133)

## What is the change?
- Upgrade Ruby to version 3.3.0
- Upgrade Bundler to version 2.5.6

## Notes

https://www.ruby-lang.org/en/news/2023/12/25/ruby-3-3-0-released/

Standard library updates

RubyGems and Bundler warn if users do require the following gems without adding them to Gemfile or gemspec. This is because they will become the bundled gems in the future version of Ruby.

Targeted libraries are:
```
abbrev
base64
bigdecimal
csv
drb
getoptlong
mutex_m
nkf
observer
racc
resolv-replace
rinda
syslog
```

[DS-4133]: https://loyaltynz.atlassian.net/browse/DS-4133?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -52,6 +52,11 @@
   #
   gem 'racksh', '~> 1.0'
 
+  gem "bigdecimal", "~> 3.1.5"
```
Comment (ildomar-grings) [inline Gemfile:55]: Any strong reasons for locking versions?

Comment (nickpiercy95) [reply Gemfile:55]: No strong reason, just figured we might as well keep the same version as what's currently included in the stdlib

Comment (ildomar-grings) [reply Gemfile:55]: I see, good catch, well observed.
Knowing those are standard libraries, I dont see them changing to represent a short-term risk.
Can we lock the major version only?

Comment (nickpiercy95) [reply Gemfile:55]: You mean like `~> 3`? That's a good call, I will do that 👍

**Review by davidoram — APPROVED** [review]
LGTM

File: Gemfile (hunk 1 above)
Comment (nickpiercy95) [reply Gemfile]: @ildomar-grings done :)

File: Gemfile [hunk 2]
```diff
@@ -29,7 +29,7 @@
   # - https://github.com/MindscapeHQ/raygun4ruby
   #
   # gem 'newrelic_rpm', '~> 5.3'
-  # gem 'ddtrace',      '~> 0.13'
+  gem 'ddtrace',      '~> 1.20.0'
```
Comment (ildomar-grings) [inline Gemfile:32]: Suggestion: `'~> 1.20'`

**Comment by ildomar-grings** [comment]
I can't approve this PR, because I started it, haha

**Pull request merged**